math/modular_inverse_simple.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file
+ * @brief simple implementation of adaption modular multiplicative inverse
+ * algorithm
+ *
+ * @details
+ * this algorithm calculate the modular inverse iteratively
+ *
+ * @ref
+ * https://en.wikipedia.org/wiki/Modular_multiplicative_inverse
+ */
+
+#include <cassert>
+#include <iostream>
+
+/**
+ * @brief Function imod
+ *  Responsible for calculating the modular inverse from x % y
+ * @param x number
+ * @param y number
+ * @returns the modular inverse
+ */
+int imod(int x, int y) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂